### PR TITLE
v1.15 Backports 2024-07-23

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #33779 (@alex-berger)
 * [x] #33823 (@tommyp1ckles)
 * [x] #33819 (@giorio94) :warning: resolved conflicts
 * [x] #33888 (@giorio94) :warning: resolved conflicts
 * [x] #33887 (@gandro)
 * [x] #33907 (@julianwiedmann)
 * [x] #33934 (@giorio94) :warning: resolved conflicts

PRs skipped due to conflicts:

 * #33803 (@mhofstetter)
 * #33903 (@chaunceyjiang)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 33779 33823 33819 33888 33887 33907 33934
```

### CODE REVIEW

**Review by giorio94 — APPROVED** [review]
My commits look good, thanks!

File: bpf/tests/conntrack_test.c [hunk 1]
```diff
@@ -215,4 +215,85 @@ int bpf_test(__maybe_unused struct __sk_buff *sctx)
...
+
+		if (!entry)
+			test_fatal("ct entry lookup failed");
+
+		seen_flags.value |= TCP_FLAG_SYN;
+
+		/* First packet is monitored */
+		res = __ct_lookup(get_ct_map4(&tuple), &ctx, &tuple,
+				  ct_tcp_select_action(seen_flags), CT_SERVICE,
+				  CT_ENTRY_SVC, &ct_state, true, seen_flags, &monitor);
```
Comment (julianwiedmann) [inline bpf/tests/conntrack_test.c]: CI noticed that the function signature in v1.15 is different. I can queue up a manual backport if you prefer, haven't checked *how* big the difference is.

Comment (julianwiedmann) [reply bpf/tests/conntrack_test.c]: `s/CT_ENTRY_SVC/CT_ENTRY_ANY` should suffice :)

Comment (joamaki) [reply bpf/tests/conntrack_test.c]: Thanks, will apply!

**Comment by joamaki** [comment]
/test-backport-1.15

**Review by squeed — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]
ty!

**Review by viktor-kurchenko — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]

**Pull request merged**